### ISSUE TITLE
Use Deep Sort to track trucks

### ISSUE THREAD

**Issue by Flock1** [nwojke/deep_sort#106]
Hey guys,
I want to know how can I use this to track trucks in a video. Should I replace this CNN with a CNN trained to detect trucks or re-train the CNN for trucks?
Let me know

**Comment by Akhtar303nu** [comment]
Hi @Flock1  go to yolo.py change the line 
if predicted_class != 'person' 
to
if predicted_class != 'truck'

**Comment by Flock1** [comment]
But I want to track it. So how do I do that using deep sort? There's no YOLO implementation in deep sort

**Comment by Akhtar303nu** [comment]
Sorry @Flock1 i have no idea about it
but I want to count vehicle(car,bus,motorbyce ) how can I count it using deep sort any I dea or suggestion
Thanks

**Comment by buzdarbalooch** [comment]
Hi,
Is there any YOLO implementation in deep sort? I have trained a model consisting (a small video) on my custom dataset. I want to track few Objects will Deep Sort be useful for that?

**Comment by Akhtar303nu** [comment]
@buzdarbalooch  Follow this https://github.com/Qidian213/deep_sort_yolov3

**Comment by nwojke** [comment]
Deep SORT uses an appearance descriptor that has been trained on a person re-identification dataset. You probably want to change that with something that has been trained on cars/trucks. If you have a dataset, the necessary steps would roughly look like this:

* Train an appearance descriptor, e.g., using [our code](https://github.com/nwojke/cosine_metric_learning)
* Generate detections using the re-trained model
* Tune Deep SORT hyperparameters

**Comment by jhonjam** [comment]
@nwojke hi i am trying to train the **cosine metric learning** model available in this repository [https://github.com/nwojke/cosine_metric_learning](url) with data set VeRi available in [https://github.com/JDAI-CV/VeRidataset](url). 
I have followed some recommendations that you described in this repository [to](https://github.com/nwojke/cosine_metric_learning/issues/5#issuecomment-381883310) carry out the training with this data set.
I have tried to carry out the training with the data set as supplied without any modification in the files, however I have the following errors.
`usage: train_veri.py [-h] [--batch_size BATCH_SIZE]
                     [--learning_rate LEARNING_RATE]
                     [--eval_log_dir EVAL_LOG_DIR]
                     [--number_of_steps NUMBER_OF_STEPS] [--log_dir LOG_DIR]
                     [--loss_mode LOSS_MODE] [--mode MODE]
                     [--restore_path RESTORE_PATH] [--run_id RUN_ID]
                     [--dataset_dir DATASET_DIR]
train_veri.py: error: unrecognized arguments: Drive/Trainig_Cosenet/VeRi`

I wanted to ask you how the file folders should be or what modifications I have to make in order to carry out this training.

tanks you

**Comment by luvwinnie** [comment]
@nwojke Hi, I have some question on this issue, I want to ask that, is it possible to train the cosine metric model for both human and vehicle classes so that the deep sort model can track both human and car/trucks and etc? Have you ever tried to train different classes with combining multiple categories?
